airbyte-db/lib/src/main/java/io/airbyte/db/check/impl/ConfigsDatabaseAvailabilityCheck.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.check.impl;
+
+import io.airbyte.db.check.DatabaseAvailabilityCheck;
+import io.airbyte.db.instance.configs.ConfigsDatabaseInstance;
+import java.util.Optional;
+import org.jooq.DSLContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of the {@link DatabaseAvailabilityCheck} for the Configurations database.
+ */
+public class ConfigsDatabaseAvailabilityCheck implements DatabaseAvailabilityCheck {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConfigsDatabaseAvailabilityCheck.class);
+
+  // TODO inject via dependency injection framework
+  private final DSLContext dslContext;
+
+  // TODO inject via dependency injection framework
+  private final long timeoutMs;
+
+  public ConfigsDatabaseAvailabilityCheck(final DSLContext dslContext, final long timeoutMs) {
+    this.dslContext = dslContext;
+    this.timeoutMs = timeoutMs;
+  }
+
+  @Override
+  public String getDatabaseName() {
+    return ConfigsDatabaseInstance.DATABASE_LOGGING_NAME;
+  }
+
+  @Override
+  public Optional<DSLContext> getDslContext() {
+    return Optional.ofNullable(dslContext);
```

Comment:
Why do we allow a to have a null dslContext? it should be either mocked or a real context? Could we use something like https://projectlombok.org/features/NonNull to validate it in the constructor? This can be compatible with the constructor annotation as well: https://projectlombok.org/features/constructor

Reply:
We shouldn't.  However, when we move to dependency injection, there is no guarantee that the `DSLContext` bean will exist.  Therefore, I am trying to be defensive here so we can raise a meaningful exception that clearly states what the issue is vs just throwing a `NullPointerException` and not being obvious.

Reply:
> We shouldn't. However, when we move to dependency injection, there is no guarantee that the DSLContext bean will exist.

This seems like it would be a recurring problem in many places, right? Does this mean we will need to wrap every bean in an optional and have explicit logic that checks if it exists and throws an error if not?

It seems like this is something that the framework should be able to enforce, i.e. bean X is required, and a meaningful exception should be thrown automatically if it does not exist at runtime. Maybe I am misunderstanding something here?